Bottom: 16b21790f90dd38c30140a70a1856f2068ac487f
Top:    03cdff31bb1c2f29df9c44caaf6e2153111b5734
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2014-07-04 04:22:00 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 5cd245f..c5cab7d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
